flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowUtil.scala

```diff
@@ -227,6 +232,11 @@ object WindowUtil {
         }
         val slide = getOperandAsLong(windowCall.operands(2))
         val size = getOperandAsLong(windowCall.operands(3))
+        if (slide <= 0 || size <= 0) {
+          throw new TableException(
+            s"HOP table function based aggregate requires slide and size being positive," +
```

Comment:
nit: being -> to be.
nit2: Just a thought we could have a separate message for each invalid value.

Reply:
> nit2: Just a thought we could have a separate message for each invalid value.

I have a different opinion about that 
1. since it will lead to the fact that once one value is fixed there will be another issue.
2. i would prefer  to have similar approach as in runtime to avoid some missed cases